cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDSpendingMonth.java

```diff
@@ -0,0 +1,97 @@
+package com.skillstorm.StepDefinitions;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.testng.Assert;
+
+import com.skillstorm.WebDriverSingleton;
+import com.skillstorm.PageObjects.LoginPage;
+import com.skillstorm.PageObjects.SpendingMonthPage;
+import com.skillstorm.PageObjects.SpendingPage;
+import com.skillstorm.Utilities.Navigator;
+import com.skillstorm.Utilities.UserData.User;
+import com.skillstorm.Utilities.UserData.UserType;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+public class SDSpendingMonth {
+    private WebDriver driver;
+    private SpendingMonthPage spendingMonthPage;
+    private SpendingPage spendingPage;
+    private Navigator navigator;
+
+    @Before("@spending-month-page")
+    public void setUp() {
+        driver = WebDriverSingleton.getDriver();
+        navigator = new Navigator(driver);
+        this.spendingMonthPage = new SpendingMonthPage(driver);
+        this.spendingPage = new SpendingPage(driver);
+    }
+
+    @After("@spending-month-page")
+    public void tearDown() {
+        if(driver != null){
+            this.driver.quit();
+        }
+    }
+
+    @Given("SpendingMonth: I am logged in")
+    public void iAmLoggedIn() {
+        navigator.navigateTo(Navigator.PGNAME_LOGIN);
+        LoginPage loginPage = new LoginPage(driver);
+        User user = new User(UserType.PERSISTANT, "frontend.tests@gmail.com", "password1");
+        loginPage.login(user);
+    }
+
+    @And("I am on the SpendingMonth page")
+    public void iAmOnTheSpendingMonthPage() {
+        this.spendingMonthPage.clickTab();
+        this.spendingPage.clickButton("See Current Month");
+        navigator.navigateTo(Navigator.PGNAME_SPENDINGMONTH);
+    }
```

Comment:
Medium Priority: I don't think it is good practice to use `this.<var>` in Step Definitions. Should just be able in this instance to use `spendingMonthPage.clickTab();`
`spendingPage.clickButton("See Current Month");`

Reply:
Updated code to remove 'this' from unnecessary occurrences in code